staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -909,6 +910,28 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 		}
 	}
 
+	if len(schema.XValidations) > 0 {
+		structural, err := structuralschema.NewStructural(schema)
+		if err != nil {
+			allErrs = append(allErrs, field.InternalError(fldPath.Child("x-kubernetes-validations"), err))
```

Comment:
This leads to duplicated errors for non-structural schemas.

Do we care? This would mean we have an old CRD here that survived from an update when v1beta1 was still a thing.

cc @liggitt

Reply:
Would this actually be an `InternalError` or a user error? If the second, you should use `field.Invalid`.

Reply:
For non-structural schemas, XValidations is blocked from adding.